daml/Marketplace/Custody/Model.daml

```diff
@@ -97,11 +108,12 @@ template LifecycleRequest
     signatory operator, provider, customer
 
     -- TODO: this shouldn't be exported; it's only called from CustodyService.Lifecycle
-    choice Process : [ ContractId AssetDeposit ]
+    choice Process : (ContractId AssetDeposit, [ContractId SettlementInstruction])
       with
           investor: Party
           safekeepingDepositCid: ContractId AssetDeposit
           fixings: Map Text (Map Date Decimal) --TODO these should be looked up from a contract on-ledger, not provided by the custodian!
+          uniquePayoutId: Text
```

Comment:
So you decided that ultimately this is provided off-ledger, right? cc @georg-da @brian-weir you agree this is ok?

Reply:
Yes.

Reply:
So the investor is expected to provide this `uniquePayoutId` as they call are calling this choice ?

Reply:
This choice is called by the "provider". That party is expected to provide a unique payout ID. That ID can be implemented in many ways, see Tests/Bond.daml - the date is used, also Tests/BinaryOption.daml - some "internal" (in this case imaginary) ID is used. One can also implement a counter contract and use that to provide the ID.

Reply:
I'm just realising that `controller investor` was already there but you've added `masterAgreementSignatories = [provider, customer, investor]`. Thus, `provider` and `investor` is actually the same party ?

Reply:
Ah I overlooked `controller investor` but was referring to the following in my previous comment:
https://github.com/digital-asset/da-marketplace/pull/301/files#diff-fbb5e9937b9cb7cb0846524c164f5290d66dc583bce10491a9ea6ac8b094e5a3R129
(see that the provider aka. bank does the lifecycling).

It seems that they can be the same party in some cases.
When defining masterAgreementSignatories, I tried to include all stakeholders (I didn't add or modify either provider, customer or investor).

Reply:
Cool 👍 

We'll tidy this up in another PR with @lucianojoublanc-da. It looks like the customer of the Custody service requests `RequestLifecycle` which creates a `Custody.LifecycleRequest` contract with the `provider` and `customer` details from the service and thus `investor: Party` on the `LifecycleRequest.Process` choice can only be party which is a signatory to that contract.